test/integration/kongaddon_test.go

```diff
@@ -0,0 +1,82 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kongaddon "github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/kong"
+	environment "github.com/kong/kubernetes-testing-framework/pkg/environments"
+)
+
+type customImageTest struct {
+	controllerImageRepo string
+	controllerImageTag  string
+	proxyImageRepo      string
+	proxyImageTag       string
+}
+
+func (tc customImageTest) controllerImage() string {
+	return fmt.Sprintf("%s:%s", tc.controllerImageRepo, tc.controllerImageTag)
+}
+
+func (tc customImageTest) proxyImage() string {
+	return fmt.Sprintf("%s:%s", tc.proxyImageRepo, tc.proxyImageTag)
+}
+
+func (tc customImageTest) name() string {
+	return fmt.Sprintf("KongAddonImages:[%s,%s]", tc.controllerImage(), tc.proxyImage())
+}
+
+func TestKongAddontWithCustomImage(t *testing.T) {
+	tests := []customImageTest{
+		{"kong/kubernetes-ingress-controller", "2.3.0", "kong", "2.7"},
+		{"kong/kubernetes-ingress-controller", "2.3.1", "kong", "2.8"},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name(), func(t *testing.T) {
+			t.Parallel()
+			testKongAddonWithCustomImage(t, tc)
```

Comment:
Any particular reason for extracting this to a separate function? I do agree with making the functions short and succinct but I wonder what's the value of putting the whole test body in a separate `func`

Reply:
Just a personal preference to keep those as separate functions.

Reply:
+1 for striving for small indentation, I'm not sure I see the benefits of using the separate functions like that.

Maybe I'll get convinced with time 🤷‍♂️